pyspecdata/lmfitdata.py

```diff
@@ -266,8 +241,12 @@ def eval(self, taxis=None, set_what=None, set_to=None):
             newdata.set_error(self.fit_axis,
                     self.get_error(self.fit_axis))
         # }}}
-        newdata
-        newdata.data[:] = self.fitfunc(p, taxis).flatten()
+        param_dict = {}
+        for j in range(len(list(p))):
+            case = {list(self.parameter_names)[j]:list(p)[j]}
+            param_dict.update(case)
+        self.set_guess(param_dict)
+        newdata.data[:] = self.run_lambda(self.pars,thistaxis = taxis).flatten()
```

Comment:
OK, so with what I suggest below, you would replace `thistaxis = taxis` in the call to `run_lambda` with `**{self.fitaxis:taxis}`

these changes are incomplete, but that way at least what we are proposing is compatible with more than one variable (most of what is/was there is compatible with only one variable, but think about fitting a 2D function like R1(τ,p))

Reply:
https://jmfrancklab.slack.com/archives/CV553TPRV/p1682685969709779?thread_ts=1682650761.090889&cid=CV553TPRV

Reply:
core of this issue is ongoing, addressed with new commit